hadoop-ozone/ozone-recon/src/main/java/org/apache/hadoop/ozone/recon/api/ContainerKeyService.java

```diff
@@ -72,10 +74,11 @@
    * @return {@link Response}
    */
   @GET
-  public Response getContainers() {
+  public Response getContainers(
+      @DefaultValue("-1") @QueryParam("limit") int limit) {
     Map<Long, ContainerMetadata> containersMap;
     try {
-      containersMap = containerDBServiceProvider.getContainers();
+      containersMap = containerDBServiceProvider.getContainers(limit);
     } catch (IOException ioEx) {
```

Comment:
I see the API to limit.
But how we shall get the list of containers from the last returned result, will be there be an API support for that?

Reply:
@bharatviswa504 This PR only supports limit param and I don't see the need for skip param support in the near feature. I will explain how the UI will consume this API to show containers and keys to the users. UI will fetch this API with an initial limit of 50 (arbitrary number or could be x% of totalCount). When the user scrolls to the end of the list, UI will trigger another call to the same API with limit + 50 to get 100 items. In a similar fashion, UI will keep loading results with infinite scroll like something similar to this demo - https://infinite-scroll.com/demo/full-page/ . Since, the results are not going to be paginated, there is no need for skip param support here in my opinion. 

The only thing missing is the totalCount in the response of these APIs and that will be implemented as part of another JIRA if needed in the future. Infinite scroll could be implemented without total count but having total count in the UI will give better user experience. 

Please let me know if you have any more questions.

Reply:
If we have support for continuation, in the next call we shall not require to fetch limit + 50, we can fetch next 50.

Reply:
Adding total count to response and supporting "start" query param is tracked via https://issues.apache.org/jira/browse/HDDS-1685 and will be implemented soon.

This PR only adds "limit" support to the APIs.